File: Primes/primes.py
import math

def main():

	prime_file = open('testprimes.txt','r')
	total = 0

	for line in prime_file.readlines():

		n = int(line)
		if is_prime(n):
			total += 1

	print total
	prime_file.close()

def is_prime(n):

	'''
	Return true if n is prime, false otherwise.
	n must be two or greater.
	'''
	assert 2 <= n, 'is_prime argument n invalid.'
	if n == 2:
		return True

	sqrt_n = int(math.sqrt(n)) + 1 # round down, add one.


	for divisor in range(2, sqrt_n):
		if n % divisor == 0: # n is not prime:
			return False

	return True

if __name__ == '__main__':
	main()

File: Primes/primes.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isPrime(int n) {
	/*
	Returns true if n is prime.
	*/

	if(n == 2){
		return 1;
	}

	int sqrtN = floor(sqrt(n)) + 1;

	for(int divisor = 2; divisor < sqrtN; divisor += 1) {
		if(n % divisor == 0) { // n is composite
			return 0;
		}
	}

	return 1;

}

int main() {

	FILE *ptr_file;

	ptr_file = fopen("testprimes.txt","r");
	int total = 0;
	char *line = NULL;
	size_t len = 0;
	ssize_t read; 
	

	while((read = getline(&line, &len, ptr_file)) != -1) {
		
		int n = atoi(line);
		if(isPrime(n)) {
			total += 1;
		}
	}

	printf("%d\n", total);
	
	fclose(ptr_file);
	return 0;
}
